projects/RabbitMQ.Client/util/NetworkOrderDeserializer.cs

```diff
@@ -43,12 +44,38 @@ internal static ushort ReadUInt16(ReadOnlySpan<byte> span)
             return BinaryPrimitives.ReadUInt16BigEndian(span);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ushort ReadUInt16(ReadOnlySequence<byte> buffer)
+        {
+            if (!BinaryPrimitives.TryReadUInt16BigEndian(buffer.First.Span, out ushort value))
+            {
+                Span<byte> bytes = stackalloc byte[2];
+                buffer.Slice(0, 2).CopyTo(bytes);
+                value = BinaryPrimitives.ReadUInt16BigEndian(bytes);
+            }
+
+            return value;
+        }
+
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static uint ReadUInt32(ReadOnlySpan<byte> span)
         {
             return BinaryPrimitives.ReadUInt32BigEndian(span);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int ReadInt32(ReadOnlySequence<byte> buffer)
+        {
+            if (!BinaryPrimitives.TryReadInt32BigEndian(buffer.First.Span, out int value))
+            {
+                Span<byte> bytes = stackalloc byte[4];
```

Comment:
Maybe I miss something. Doesn't this under the covers do `MemoryMarshal.Read` which should copy into the int? Why is the stackalloc needed?